### PR TITLE
Implemented tests for issue 715

### PR DESCRIPTION
**Pull request by jcalderin**
Refactor the steps and the mock. Now, the lettuce_tools mock version is used to simulate the Context Provider funtionality

### CODE REVIEW

File: test/acceptance/integration/ISSUE-715 CPr-propagate X-Auth-Token header in forwarded requests/__init__.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+# Copyright 2013 Telefonica Investigacion y Desarrollo, S.A.U
```
Comment (fgalan) [inline test/acceptance/integration/ISSUE-715 CPr-propagate X-Auth-Token header in forwarded requests/__init__.py]: New file... I'd suggest to use 2015.

File: test/acceptance/integration/ISSUE-715 CPr-propagate X-Auth-Token header in forwarded requests/issue-715.feature [hunk 2]
```diff
@@ -0,0 +1,157 @@
+# -*- coding: utf-8 -*-
+# Copyright 2013 Telefonica Investigacion y Desarrollo, S.A.U
```
Comment (fgalan) [inline test/acceptance/integration/ISSUE-715 CPr-propagate X-Auth-Token header in forwarded requests/issue-715.feature]: New file... I'd suggest to use 2015.

File: test/acceptance/integration/ISSUE-715 CPr-propagate X-Auth-Token header in forwarded requests/issue-715.feature [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    And build the standard entity update payload with the previous data
+    When a standard context entity update is asked with the before information
+    #Mock information
+    Then retrieve information from the mock
+    And headers of the last mock petition not contains the head "X-Auth-Token"
+    And the path in the last mock petition contains "service"
+    And there is "1" petitions requested to the mock
+    And clean the mongo database of the service "issue_715"
+
+#Fixme: Repeat all tests with convenience operation, when developed
```
Comment (fgalan) [inline test/acceptance/integration/ISSUE-715 CPr-propagate X-Auth-Token header in forwarded requests/issue-715.feature]: Maybe we could add more precission to this, e.g.

"#Fixme: Repeat all tests with convenience operation, when developed. This is pending on the "big refactoring" in CPr funciontality being done in 1Q2015"

Comment (jcalderin) [inline test/acceptance/integration/ISSUE-715 CPr-propagate X-Auth-Token header in forwarded requests/issue-715.feature]: fixed in 250707c

File: test/acceptance/integration/ISSUE-715 CPr-propagate X-Auth-Token header in forwarded requests/issue-715.feature (hunk 2 above)
Comment (jcalderin) [inline test/acceptance/integration/ISSUE-715 CPr-propagate X-Auth-Token header in forwarded requests/issue-715.feature]: Fixed in 250707c

File: test/acceptance/integration/ISSUE-715 CPr-propagate X-Auth-Token header in forwarded requests/__init__.py (hunk 1 above)
Comment (jcalderin) [inline test/acceptance/integration/ISSUE-715 CPr-propagate X-Auth-Token header in forwarded requests/__init__.py]: Fixed in 250707c

**Comment by fgalan** [comment]
LGTM (passing the ball to @iariasleon for final LGTM + merge + delete branch)

File: test/acceptance/integration/ISSUE-715 CPr-propagate X-Auth-Token header in forwarded requests/issue-715.feature [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
```
Comment (iariasleon) [inline test/acceptance/integration/ISSUE-715 CPr-propagate X-Auth-Token header in forwarded requests/issue-715.feature]: the __author__ line is missing

File: test/acceptance/integration/steps_lib/general.py [hunk 5]
```diff
@@ -59,6 +59,18 @@ def a_new_ngsi10_petition_with_the_service_and_the_subservice(step, ngsi_version
         raise ValueError(
             'The version of ngsi api have to be \'9\' or \'10\', not {ngsi_version}'.format(ngsi_version=ngsi_version))
 
+@step('add the following headers to the petition')
```
Comment (iariasleon) [inline test/acceptance/integration/steps_lib/general.py]: using unicode string is better, see:
https://docs.python.org/2/tutorial/introduction.html#unicode-strings

File: test/acceptance/integration/steps_lib/mock_utils.py [hunk 6]
```diff
@@ -71,7 +71,6 @@ def the_path_in_the_last_petition_contains(step, head, value):
     mock_response = eval(world.mock_data.text)['requests']
```
Comment (iariasleon) [inline test/acceptance/integration/steps_lib/mock_utils.py]: why do not use constants? In 'requests' case

Comment (iariasleon) [inline test/acceptance/integration/steps_lib/mock_utils.py]: why do not use constants? In 'requests' case

File: test/acceptance/integration/ISSUE-715 CPr-propagate X-Auth-Token header in forwarded requests/issue-715.feature (hunk 4 above)
Comment (jcalderin) [inline test/acceptance/integration/ISSUE-715 CPr-propagate X-Auth-Token header in forwarded requests/issue-715.feature]: Fixed in 6b81f46

**Comment by iariasleon** [comment]
I recomend to create a README.md in fiware-orion/test/acceptance with some explication about these tests

**Comment by jcalderin** [comment]
NTC the Readme is created in another branch, i hope commit it from today to tomorrow

File: test/acceptance/integration/steps_lib/mock_utils.py (hunk 6 above)
Comment (jcalderin) [inline test/acceptance/integration/steps_lib/mock_utils.py]: Its not necessary, but i will study the use of constante in the next implementation
NTC

File: test/acceptance/integration/steps_lib/general.py (hunk 5 above)
Comment (jcalderin) [inline test/acceptance/integration/steps_lib/general.py]: I prefer not to use unicode, but i will read the link to adapt to the best practice
NTC

**Comment by iariasleon** [comment]
LGTM

**Pull request merged**